invocation.md

```diff
@@ -0,0 +1,1170 @@
+  "do": "msg/send",
+  "input": {
+    "to": ["boris@example.com", "carol@example.com"],
+    "subject": "Coffee",
+    "body": "Hey you two, I'd love to get coffee sometime and talk about UCAN Tasks!"
+  },
+  "meta": {
+    "dev/tags": ["friends", "coffee"],
+    "dev/priority": "high"
+  },
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  },
+  "prf": [
+    { "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a" }
+  ]
+}
+```
+
+Running WebAssembly from binary:
+
+```json
+{
+  "v": "0.1.0",
+  "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+  "aud": "did:web:ucan.run",
+  "with": "data:application/wasm;base64,AHdhc21lci11bml2ZXJzYWwAAAAAAOAEAAAAAAAAAAD9e7+p/QMAkSAEABH9e8GowANf1uz///8UAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP////8AAAAACAAAACoAAAAIAAAABAAAACsAAAAMAAAACAAAANz///8AAAAA1P///wMAAAAlAAAALAAAAAAAAAAUAAAA/Xu/qf0DAJHzDx/44wMBqvMDAqphAkC5YAA/1mACALnzB0H4/XvBqMADX9bU////LAAAAAAAAAAAAAAAAAAAAAAAAAAvVXNlcnMvZXhwZWRlL0Rlc2t0b3AvdGVzdC53YXQAAGFkZF9vbmUHAAAAAAAAAAAAAAAAYWRkX29uZV9mAAAADAAAAAAAAAABAAAAAAAAAAkAAADk////AAAAAPz///8BAAAA9f///wEAAAAAAAAAAQAAAB4AAACM////pP///wAAAACc////AQAAAAAAAAAAAAAAnP///wAAAAAAAAAAlP7//wAAAACM/v//iP///wAAAAABAAAAiP///6D///8BAAAAqP///wEAAACk////AAAAAJz///8AAAAAlP///wAAAACM////AAAAAIT///8AAAAAAAAAAAAAAAAAAAAAAAAAAET+//8BAAAAWP7//wEAAABY/v//AQAAAID+//8BAAAAxP7//wEAAADU/v//AAAAAMz+//8AAAAAxP7//wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAU////pP///wAAAAAAAQEBAQAAAAAAAACQ////AAAAAIj///8AAAAAAAAAAAAAAADQAQAAAAAAAA==",
+  "do": "wasm/run",
+  "input": {
+    "func": "add_one",
+    "args": [42]
+  },
+  "meta": {
+    "dev/notes": "The standard Wasm demo",
+    "ipvm/verification": "attestation",
+    "ipvm/resources": {
+      "gas": 5000
+    }
+  },
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  },
+  "prf": [
+    { "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a" }
+  ]
+}
+```
+
+Batch invocation is simply passing promises as inputs to an invocation
+
+```json
+{
+  "v": "0.1.0",
+  "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+  "aud": "did:web:ucan.run",
+  "with": "javascript:(data) => data",
+  "do": "js/call",
+  "input": {
+    "publishPost": {
+      "<-": [
+        {
+          "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+          "aud": "did:web:ucan.run",
+          "with": "https://example.com/blog/posts",
+          "do": "crud/create",
+          "input": {
+            "headers": {
+              "content-type": "application/json"
+            },
+            "payload": {
+              "title": "How UCAN Tasks Changed My Life",
+              "body": "This is the story of how one spec changed everything...",
+              "topics": ["authz", "journal"],
+              "draft": true
+            }
+          },
```

Comment:
At least for IPVM, we're a hard no on this layout

Reply:
Actually, maybe another way: if this is here purely for IPVM's benefit, I think we can get rid of this nested format completely. If we align on putting `Auth` in the canonicalization for `Promise`s, IIRC the reason for this that you had initially used this as a solve for probably just goes away.

Reply:
Yeah I was just trying to show that you could have named tasks without building batches into spec, if links are fine for IPVM I'd gladly remove inlining.